experimental/vjn/yarpl/include/yarpl/flowable.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+#include <reactivestreams/publisher.h>
+#include <reactivestreams/subscription.h>
+#include <reactivestreams/type_traits.h>
+
+#include "yarpl/scheduler.h"
+
+namespace yarpl {
+
+template <typename T>
+class Flowable : public reactivestreams::Publisher<T>,
+                 public std::enable_shared_from_this<Flowable<T>> {
+  using Subscriber = reactivestreams::Subscriber<T>;
+  using Subscription = reactivestreams::Subscription<T>;
+
+ public:
+  template <typename F,
+            typename = typename std::enable_if<
+                std::is_callable<F(Subscriber&), void>::value>::type>
+  static std::shared_ptr<Flowable> create(F&& function) {
```

Comment:
to implement proper perfect forwarding, the parameter should be of type F and then std::forward is appropriate.